internal/subscriber/core_test.go

```diff
@@ -0,0 +1,243 @@
+			if err == nil {
+				assert.True(t, EqualOnly(sub, tt.expected, []string{"topic", "subscriberID", "requestChan", "ackChan", "modAckChan"}))
+			}
+			if err != nil && !tt.wantErr {
+				t.Errorf("NewSubscriber Error %v", err)
+			}
+		})
+	}
+}
+
+func TestNewCore(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockBrokerStoreCore := mocks.NewMockIBrokerStore(ctrl)
+	mockSubscriptionCore := sCore.NewMockICore(ctrl)
+	mockOffsetCore := oCore.NewMockICore(ctrl)
+	mockCache := mCache.NewMockICache(ctrl)
+
+	type args struct {
+		bs               brokerstore.IBrokerStore
+		subscriptionCore subscription.ICore
+		offsetCore       offset.ICore
+		ch               cache.ICache
+	}
+	tests := []struct {
+		name string
+		args args
+		want ICore
+	}{
+		{
+			name: "Get New subscriber Core",
+			args: args{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+			want: &Core{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, NewCore(tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch), "NewCore(%v, %v, %v, %v)", tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch)
+		})
+	}
+}
+
+func getMockConsumerCore(ctx context.Context, ctrl *gomock.Controller) *mockMB.MockConsumer {
+	cs := mockMB.NewMockConsumer(ctrl)
+	req := messagebroker.GetTopicMetadataRequest{
+		Topic:     topicName,
+		Partition: partition,
+	}
+	cs.EXPECT().GetTopicMetadata(ctx, req).Return(messagebroker.GetTopicMetadataResponse{}, nil).AnyTimes()
+	cs.EXPECT().Pause(ctx, gomock.Any()).AnyTimes().Return(nil)
+	cs.EXPECT().CommitByPartitionAndOffset(gomock.Any(), gomock.Any()).Return(
+		messagebroker.CommitOnTopicResponse{}, nil,
+	).AnyTimes()
+	return cs
+}
+
+func getDummySubscription() subscription.Model {
+	return subscription.Model{
+		Name:  subName,
+		Topic: topicName,
+		PushConfig: &subscription.PushConfig{
+			PushEndpoint: "https://www.razorpay.com/api",
+			Attributes: map[string]string{
+				"test_key": "test_value",
+			},
+			Credentials: &credentials.Model{
+				Username: "test_user",
+				Password: "",
+			},
+		},
+		DeadLetterPolicy:               nil,
+		RetryPolicy:                    nil,
+		AckDeadlineSeconds:             10,
+		ExtractedTopicProjectID:        "project123",
+		ExtractedTopicName:             topicName,
+		ExtractedSubscriptionName:      subName,
+		ExtractedSubscriptionProjectID: "project123",
+	}
+}
+
+func EqualOnly(sub1 interface{}, sub2 interface{}, ExceptField []string) bool {
```

Comment:
this seems like a util func to me, can we move this to a separate package

Reply:
moved to utils package